python/examples/llama/test_llama3.py

```diff
@@ -0,0 +1,1889 @@
+    else:
+        keys = xk_rot
+        values = xv
+
+    # Transpose for attention: (batch, n_heads, seq_len, head_dim)
+    xq_t_shape = [batch, n_heads, seq_len, head_dim]
+    xq_t = tensor.EmptyOp(xq_t_shape, elty).result
+
+    # Permute [0, 2, 1, 3]
+    d0, d1, d2, d3 = [ir.AffineDimExpr.get(i) for i in range(4)]
+    xq_perm_map = affine_map(4, [d0, d2, d1, d3])
+    xq_t_map = affine_map(4, [d0, d1, d2, d3])
+
+    @linalg.generic(
+        [xq_rot],
+        [xq_t],
+        [xq_perm_map, xq_t_map],
+        [parallel] * 4,
+    )
+    def transpose_xq(val, _out):
+        return val
+
+    xq_transposed = transpose_xq
+
+    # Transpose keys and values similarly
+    keys_t = tensor.EmptyOp(xq_t_shape, elty).result
+
+    @linalg.generic(
+        [keys],
+        [keys_t],
+        [xq_perm_map, xq_t_map],
+        [parallel] * 4,
+    )
+    def transpose_k(val, _out):
+        return val
+
+    keys_transposed = transpose_k
+
+    values_t = tensor.EmptyOp(xq_t_shape, elty).result
+
+    @linalg.generic(
+        [values],
+        [values_t],
+        [xq_perm_map, xq_t_map],
+        [parallel] * 4,
+    )
+    def transpose_v(val, _out):
+        return val
+
+    values_transposed = transpose_v
+
+    # Compute attention scores: matmul(xq, keys.transpose(-2, -1))
+    # xq_transposed: (batch, n_heads, seq_len, head_dim)
+    # keys_transposed: (batch, n_heads, seq_len, head_dim) -> transpose to (batch, n_heads, head_dim, seq_len)
+    # scores: (batch, n_heads, seq_len, seq_len)
+    scores_shape = [batch, n_heads, seq_len, seq_len]
+    scores_uninit = tensor.EmptyOp(scores_shape, elty).result
+    scores_zeroed = linalg.fill(zero, outs=[scores_uninit])
+
+    # Batched matmul with transpose
+    b, h, s1, s2, d = [ir.AffineDimExpr.get(i) for i in range(5)]
+    xq_scores_map = affine_map(5, [b, h, s1, d])
+    keys_scores_map = affine_map(5, [b, h, s2, d])  # Will read from transposed position
+    scores_map = affine_map(5, [b, h, s1, s2])
+
+    @linalg.generic(
+        [xq_transposed, keys_transposed],
+        [scores_zeroed],
+        [xq_scores_map, keys_scores_map, scores_map],
+        [parallel, parallel, parallel, parallel, reduction],
+    )
+    def compute_scores(q_val, k_val, score_val):
+        prod = arith.MulFOp(q_val, k_val).result
+        return arith.AddFOp(score_val, prod).result
+
+    scores_raw = compute_scores
+
+    # Scale by 1/sqrt(head_dim)
+    scale_val = 1.0 / pymath.sqrt(head_dim)
+    scale_const = arith.constant(elty, scale_val)
+    scores_scaled_uninit = tensor.EmptyOp(scores_shape, elty).result
+
+    d0, d1, d2, d3 = [ir.AffineDimExpr.get(i) for i in range(4)]
+    identity_map = affine_map(4, [d0, d1, d2, d3])
+
+    @linalg.generic(
+        [scores_raw],
+        [scores_scaled_uninit],
+        [identity_map, identity_map],
+        [parallel] * 4,
+    )
+    def scale_scores(score, _out):
+        return arith.MulFOp(score, scale_const).result
```

Comment:
```suggestion
        return arith.mulf(score, scale_const)
```
If you use `arith.mulf` (or whatever the snake_case version is called), you should be able to elide the `.result`.

This holds generally (for single-result ops, that is).